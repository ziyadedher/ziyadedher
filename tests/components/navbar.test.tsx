/**
 * @jest-environment jsdom
 */

// eslint-disable-next-line import/no-unassigned-import -- @testing-library/jest-dom doesn't need to be bound.
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import Navbar, { NavbarPage } from "../../src/components/navbar";

describe("navbar", () => {
  test("renders when current page is HOME", () => {
    const result = render(<Navbar currentPage={NavbarPage.HOME} />);
    expect(result).toMatchSnapshot();
  });

  test("renders when current page is HACKS", () => {
    const result = render(<Navbar currentPage={NavbarPage.HACKS} />);
    expect(result).toMatchSnapshot();
  });
});
