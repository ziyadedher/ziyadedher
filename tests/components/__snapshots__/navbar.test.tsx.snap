// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar renders when current page is HACKS 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="flex flex-grow justify-center space-x-4 w-full"
      >
        <a
          href="/"
        >
          <span
            class="uppercase font-light text-sm hover:opacity-50 transition-all opacity-100"
          >
            Home
          </span>
        </a>
        <a
          href="/hacks"
        >
          <span
            class="uppercase font-light text-sm hover:opacity-50 transition-all opacity-70"
          >
            Hacks
          </span>
        </a>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="flex flex-grow justify-center space-x-4 w-full"
    >
      <a
        href="/"
      >
        <span
          class="uppercase font-light text-sm hover:opacity-50 transition-all opacity-100"
        >
          Home
        </span>
      </a>
      <a
        href="/hacks"
      >
        <span
          class="uppercase font-light text-sm hover:opacity-50 transition-all opacity-70"
        >
          Hacks
        </span>
      </a>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`navbar renders when current page is HOME 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="flex flex-grow justify-center space-x-4 w-full"
      >
        <a
          href="/"
        >
          <span
            class="uppercase font-light text-sm hover:opacity-50 transition-all opacity-70"
          >
            Home
          </span>
        </a>
        <a
          href="/hacks"
        >
          <span
            class="uppercase font-light text-sm hover:opacity-50 transition-all opacity-100"
          >
            Hacks
          </span>
        </a>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="flex flex-grow justify-center space-x-4 w-full"
    >
      <a
        href="/"
      >
        <span
          class="uppercase font-light text-sm hover:opacity-50 transition-all opacity-70"
        >
          Home
        </span>
      </a>
      <a
        href="/hacks"
      >
        <span
          class="uppercase font-light text-sm hover:opacity-50 transition-all opacity-100"
        >
          Hacks
        </span>
      </a>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
